test/unit/puppetlabs/services/master/master_core_test.clj

```diff
@@ -189,6 +189,29 @@
                         "supported"]
                        "bogus"}]})))))))
 
+(deftest valid-static-file-path-test
+    (let [valid-paths ["modules/foo/files/bar.txt"
+                       "modules/foo/files/bar/"
+                       "modules/foo/files/bar/baz.txt"
+                       "environments/production/modules/foo/files/bar.txt"
+                       "environments/production/modules/foo/files/..conf..d../..bar..txt.."
+                       "environments/test/modules/foo/files/bar/baz.txt"
+                       "environments/dev/modules/foo/files/path/to/file/something.txt"]
+          invalid-paths ["modules/foo/manifests/bar.pp"
+                         "manifests/site.pp"
+                         "environments/foo/bar/files"
+                         "environments/../manifests/files/site.pp"
+                         "environments/../modules/foo/lib/puppet/parser/functions/site.rb"
+                         "environments/production/files/~/.bash_profile"
+                         "environments/../modules/foo/files/site.pp"
+                         "environments/production/modules/foo/files/../../../../../../site.pp"]]
```

Comment:
Removing one seems reasonable to me. The unit seems like it should the one to stay then.